sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -0,0 +1,197 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.command
+
+import org.apache.spark.Logging
+import org.apache.spark.sql.{Row, SQLContext}
+import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.catalyst.expressions.{Attribute, AttributeReference}
+import org.apache.spark.sql.execution.datasources.BucketSpec
+import org.apache.spark.sql.types._
+
+
+/**
+ * A DDL command expected to be run in the underlying system without Spark parsing the
+ * query text.
+ */
+abstract class NativeDDLCommands(val sql: String) extends RunnableCommand {
+
+  override def run(sqlContext: SQLContext): Seq[Row] = {
+    sqlContext.runNativeSql(sql)
+  }
+
+  override val output: Seq[Attribute] = {
+    Seq(AttributeReference("result", StringType, nullable = false)())
+  }
+
+}
+
+case class CreateDatabase(
+    databaseName: String,
+    allowExisting: Boolean,
+    path: Option[String],
+    comment: Option[String],
+    props: Map[String, String])(sql: String)
+  extends NativeDDLCommands(sql) with Logging
+
+case class CreateFunction(
+    functionName: String,
+    alias: String,
+    resourcesMap: Map[String, String],
+    isTemp: Boolean)(sql: String)
+  extends NativeDDLCommands(sql) with Logging
+
+case class AlterTableRename(
+    tableName: TableIdentifier,
+    renameTableName: TableIdentifier)(sql: String)
+  extends NativeDDLCommands(sql) with Logging
+
+case class AlterTableSetProperties(
+    tableName: TableIdentifier,
+    setProperties: Map[String, Option[String]])(sql: String)
+  extends NativeDDLCommands(sql) with Logging
+
+case class AlterTableDropProperties(
+    tableName: TableIdentifier,
+    dropProperties: Map[String, Option[String]],
+    allowExisting: Boolean)(sql: String)
+  extends NativeDDLCommands(sql) with Logging
+
+case class AlterTableSerDeProperties(
+    tableName: TableIdentifier,
+    serdeClassName: Option[String],
+    serdeProperties: Option[Map[String, Option[String]]],
+    partition: Option[Map[String, Option[String]]])(sql: String)
+  extends NativeDDLCommands(sql) with Logging
+
+case class AlterTableStoreProperties(
+    tableName: TableIdentifier,
+    buckets: Option[BucketSpec],
+    // TODO: use `clustered` and `sorted` instead for simplicity
+    noClustered: Boolean,
+    noSorted: Boolean)(sql: String)
+  extends NativeDDLCommands(sql) with Logging
+
+case class AlterTableSkewed(
+    tableName: TableIdentifier,
+    skewedCols: Seq[String],
+    skewedValues: Seq[Seq[String]],
+    storedAsDirs: Boolean,
+    notSkewed: Boolean,
+    // TODO: what??
+    notStoredAsDirs: Boolean)(sql: String)
```

Comment:
@viirya I'm really confused about this flag. We have another one called `storedAsDirs` and this one says `notStoredAsDirs`. It would seem that one is always the opposite of the other, but that's not actually the case in tests, e.g.

```
// DDLCommandSuite, "alter table: skewed"
val expected3 = AlterTableSkewed(
  TableIdentifier("table_name", None),
  Seq("dt", "country"),
  Seq(List("2008-08-08", "us"), List("2009-09-09", "uk")),
  storedAsDirs = false,
  notSkewed = false,
  notStoredAsDirs = false)(sql3)
```
seems contradictory?